From c6a14e29c6c2a279742a13ed5a827962d206dda6 Mon Sep 17 00:00:00 2001
From: Petr Oros <poros@redhat.com>
Date: Thu, 23 Oct 2014 12:22:24 -0400
Subject: [redhat] Makefile: add -r param for show-kabi

Message-id: <7e579994746bf0b81b6f690cc721973e8af8fc28.1414066546.git.poros@redhat.com>
Patchwork-id: 98311
O-Subject: [RHEL7.1 PATCHv2 1/2] makefile: add -r param for show-kabi
Bugzilla:
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

show-kabi merging files from source into one output file
        when use -m param for generate Module.kabi_ARCH file,
        show-kabi need also -r for generate latest Module.kabi_ARCH file.
        without -r script also adding removed symbols and kernel build fail

Signed-off-by: Petr Oros <poros@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/Makefile b/redhat/Makefile
index 8639344..e442de1 100644
--- a/redhat/Makefile
+++ b/redhat/Makefile
@@ -34,7 +34,7 @@ default: rh-help
 rh-kabi:
  @for KABIARCH in "ppc64" "x86_64" "s390x"; do \
   $(REDHAT)/kabi/show-kabi -k $(REDHAT)/kabi/kabi-module/ -m \
-   -a $$KABIARCH > $(REDHAT)/kabi/Module.kabi_$$KABIARCH;\
+   -a $$KABIARCH -r $(RHELMAJOR).$(RHELMINOR) > $(REDHAT)/kabi/Module.kabi_$$KABIARCH;\
   for i in {0..$(RHELMINOR)}; do \
    mkdir -p $(REDHAT)/kabi/kabi-rhel$(RHELMAJOR)$$i/;\
    $(REDHAT)/kabi/show-kabi -k $(REDHAT)/kabi/kabi-module/ -s -a $$KABIARCH \
-- 
1.7.1